projects/packages/google-analytics/src/class-legacy.php

```diff
@@ -0,0 +1,344 @@
+				$track[ $k ] = preg_replace( '/[^a-z0-9\.\/\+\?=-]+/i', '_', $track[ $k ] );
+			}
+
+			$track[ $k ] = trim( $track[ $k ], '_' );
+		}
+		$char = ( strpos( $track['data'], '?' ) === false ) ? '?' : '&amp;';
+		return str_replace( "'", "\'", "/{$track['code']}/{$track['data']}{$char}referer=" . rawurlencode( isset( $_SERVER['HTTP_REFERER'] ) ? esc_url_raw( wp_unslash( $_SERVER['HTTP_REFERER'] ) ) : '' ) );
+	}
+
+	/**
+	 * This injects the Google Analytics code into the footer of the page.
+	 * Called exclusively by wp_head action
+	 */
+	public function insert_code() {
+		$tracking_id = Options::get_tracking_code();
+		if ( empty( $tracking_id ) ) {
+			echo "<!-- Your Google Analytics Plugin is missing the tracking ID -->\r\n";
+			return;
+		}
+
+		// If we're in the admin_area or DNT is honored and enabled, return without inserting code.
+		if (
+			is_admin()
+			|| Utils::is_dnt_enabled()
+		) {
+			return;
+		}
+
+		// @phan-suppress-next-line PhanUndeclaredClassMethod
+		if ( class_exists( 'Jetpack_AMP_Support' ) && \Jetpack_AMP_Support::is_amp_request() ) {
+			// For Reader mode — legacy.
+			add_filter( 'amp_post_template_analytics', array( GA_Manager::class, 'amp_analytics_entries' ), 1000 );
+			// For Standard and Transitional modes.
+			add_filter( 'amp_analytics_entries', array( GA_Manager::class, 'amp_analytics_entries' ), 1000 );
+			return;
+		}
+
+		if ( str_starts_with( $tracking_id, 'G-' ) ) {
+			$this->render_gtag_code( $tracking_id );
+		} else {
+			$this->render_ga_code( $tracking_id );
+		}
+	}
+
+	/**
+	 * Renders legacy ga.js code.
+	 *
+	 * @param string $tracking_id Google Analytics measurement ID.
+	 */
+	private function render_ga_code( $tracking_id ) {
+		$custom_vars = array(
+			"_gaq.push(['_setAccount', '{$tracking_id}']);",
+		);
+
+		$track = array();
+		if ( is_404() ) {
+			// This is a 404 and we are supposed to track them.
+			$custom_vars[] = "_gaq.push(['_trackEvent', '404', document.location.href, document.referrer]);";
+		} elseif (
+			is_search()
+			&& isset( $_REQUEST['s'] ) // phpcs:ignore WordPress.Security.NonceVerification.Recommended -- Function renders client-side JS, no site actions.
+		) {
+			// Set track for searches, if it's a search, and we are supposed to.
+			$track['data'] = sanitize_text_field( wp_unslash( $_REQUEST['s'] ) ); // phpcs:ignore WordPress.Security.NonceVerification.Recommended -- Function renders client-side JS, no site actions.
+			$track['code'] = 'search';
+		}
+
+		if ( ! empty( $track ) ) {
+			$track['url'] = $this->get_url( $track );
+			// adjust the code that we output, account for both types of tracking.
+			$track['url']  = esc_js( str_replace( '&', '&amp;', $track['url'] ) );
+			$custom_vars[] = "_gaq.push(['_trackPageview','{$track['url']}']);";
+		} else {
+			$custom_vars[] = "_gaq.push(['_trackPageview']);";
+		}
+
+		/**
+		 * Allow for additional elements to be added to the classic Google Analytics queue (_gaq) array
+		 *
+		 * @since 5.4.0
```

Comment:
I think we should prefix those existing `@since` with Jetpack, so in the future it's clear where it's coming from and when it was added.

```suggestion
		 * @since jetpack-5.4.0
```